rclpy/rclpy/node.py

```diff
@@ -217,12 +231,109 @@ def get_logger(self):
         """Get the nodes logger."""
         return self._logger
 
+    def declare_parameter(
+        self,
+        name: str,
+        value: ParameterValue,
+        descriptor: ParameterDescriptor
```

Comment:
I think it makes sense to make `value` and `descriptor` optional, in case we just want to declare the names of parameters and set the value/descriptor later. I believe this is more akin to `rclcpp`.

e.g.

```python
def declare_parameter(
        self,
        name: str,
        value: ParameterValue = ParameterValue(),
        descriptor: ParameterDescriptor = ParameterDescriptor()
```

Or perhaps `None`.

Reply:
Addressed in https://github.com/ros2/rclpy/pull/325/commits/0925fe4e4a72a4b32ff1ec58790d73c4ba35838a.